src/community/leiden.c

```diff
@@ -1263,3 +1271,239 @@ igraph_error_t igraph_community_leiden(
+ *    for directed graphs. Edge weights are allowed to be negative.
+ *    \cli IGRAPH_LEIDEN_OBJECTIVE_ER
+ *    Use an objective function based on the Erdős-Rényi G(n,p) null model,
+ *    defined as
+ *    <code>Q = 1/(2m) sum_ij (A_ij - γ p) δ(c_i, c_j)</code>
+ *    for undirected graphs and as
+ *    <code>Q = 1/m sum_ij (A_ij - γ) δ(c_i, c_j)</code>
+ *    for directed graphs. \c p is the weighted density, i.e. the average
+ *    link strength between all vertex pairs (whether adjacent or not).
+ *    It is computed with the assumption that self-loops cannot be present
+ *    in the graph. If self-loops are present, an error is raised.
+ *    Edge weights must not be negative.
+ *    \cli IGRAPH_LEIDEN_OBJECTIVE_ER_LOOPS
+ *    Same as \c IGRAPH_LEIDEN_OBJECTIVE_ER but computes the density with
+ *    the assumption that self-loops may be present in the graph.
+ *    \endclist
+ *    In the above formulas, \c A is the adjacency matrix, \c m is the total
+ *    edge weight, \c k are the (out- and in-) degrees, \c γ is the resolution
+ *    parameter, and <code>δ(c_i, c_j)</code> is 1 if vertices \c i and \c j
+ *    are in the same community and 0 otherwise.
+ * \param resolution The resolution parameter, which is represented by γ in
+ *    the objective functions detailed above.
+ * \param beta The randomness used in the refinement step when merging. A small
+ *    amount of randomness (\c beta = 0.01) typically works well.
+ * \param start Start from membership vector. If this is true, the optimization
+ *    will start from the provided membership vector. If this is false, the
+ *    optimization will start from a singleton partition.
+ * \param n_iterations Iterate the core Leiden algorithm the indicated number
+ *    of times. If this is a negative number, it will continue iterating until
+ *    an iteration did not change the clustering. Two iterations are often
+ *    sufficient, thus 2 is a reasonable default.
+ * \param membership The membership vector. If \p start is set to \c false,
+ *    it will be resized appropriately. If \p start is \c true, it must be
+ *    a valid membership vector for the given \p graph.
+ * \param nb_clusters The number of clusters contained in the final \p membership.
+ *    If \c NULL, the number of clusters will not be returned.
+ * \param quality The quality of the partition, in terms of the objective
+ *    function selected by \p objective. If \c NULL the quality will
+ *    not be calculated.
+ * \return Error code.
+ *
+ * Time complexity: near linear on sparse graphs.
+ *
+ * \sa \ref igraph_community_leiden() for a more flexible interface that
+ * allows specifying raw vertex weights.
+ */
+igraph_error_t igraph_community_leiden_simple(
+        const igraph_t *graph,
+        const igraph_vector_t *weights,
+        igraph_leiden_objective_t objective,
+        igraph_real_t resolution,
+        igraph_real_t beta,
+        igraph_bool_t start,
```

Comment:
If this is meant to be a simpler interface, I wonder whether we should get rid of the `start` argument and just start the algorithm from the singleton clustering unconditionally. How often do people want to use a non-singleton clustering as a starting point?